src/app/component/shared/detail-view/detail-view.component.spec.ts

```diff
@@ -1,23 +1,214 @@
 import { async, ComponentFixture, TestBed } from "@angular/core/testing";
+import { AbstractModel } from "src/app/models/AbstractModel";
 import { DetailViewComponent } from "./detail-view.component";
+import { RenderViewComponent } from "./view/view.component";
+
+class MockModel extends AbstractModel {
+  constructor(opts: any) {
+    super(opts);
+  }
+
+  public redirectPath(...args: any): string {
+    throw new Error("Method not implemented.");
+  }
+  public toJSON(): object {
+    throw new Error("Method not implemented.");
+  }
+}
 
 describe("DetailViewComponent", () => {
   let component: DetailViewComponent;
   let fixture: ComponentFixture<DetailViewComponent>;
 
+  function getWrapper() {
+    return (fixture.nativeElement as HTMLElement).querySelector("div");
+  }
+
+  function getLabels() {
+    return (fixture.nativeElement as HTMLElement).querySelectorAll("dt");
+  }
+
+  function getAnswers() {
+    return (fixture.nativeElement as HTMLElement).querySelectorAll("dl");
+  }
+
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      declarations: [DetailViewComponent],
+      declarations: [DetailViewComponent, RenderViewComponent],
     }).compileComponents();
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(DetailViewComponent);
     component = fixture.componentInstance;
-    fixture.detectChanges();
+  });
+
+  afterAll(() => {
+    viewport.reset();
   });
 
   it("should create", () => {
+    fixture.detectChanges();
     expect(component).toBeTruthy();
   });
+
+  describe("details", () => {
+    it("should handle missing details", () => {
+      fixture.detectChanges();
+      expect(getLabels().length).toBe(0);
+    });
+
+    it("should handle empty details", () => {
+      component.fields = [];
+      fixture.detectChanges();
+      expect(getLabels().length).toBe(0);
+    });
+
+    describe("single list detail", () => {
+      beforeEach(() => {
+        component.fields = [
+          {
+            key: "id",
+            templateOptions: {
+              label: "custom label",
+            },
+          },
+        ];
+        component.model = new MockModel({ id: 0 });
+        fixture.detectChanges();
+      });
+
+      it("should handle single detail", () => {
+        expect(getLabels().length).toBe(1);
+      });
+
+      it("should handle single answer", () => {
+        expect(getAnswers().length).toBe(1);
+      });
+
+      it("should create label", () => {
+        const labels = getLabels();
+        expect(labels[0].innerText.trim()).toBe("custom label");
+      });
+
+      it("should create answer", () => {
+        const answers = getAnswers();
+        expect(answers[0].innerText.trim()).toBe("0");
+      });
```

Comment:
bad terminology here.

it should be `field` and `value` not `question` and `answer`